tsdb/db.go

```diff
@@ -583,6 +583,29 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 	if err := removeBestEffortTmpDirs(l, dir); err != nil {
 		return nil, errors.Wrap(err, "remove tmp dirs")
 	}
+	if opts.WALSegmentSize >= 0 {
```

Comment:
Why this condition?

Reply:
So that truncation happens only when wal is enabled. Should we truncate irrespective of whether wal is enabled or disabled?

Reply:
Ah, I'd incorrectly presumed this was the flag value.